packages/@tinacms/cli/src/next/config-manager.ts

```diff
@@ -0,0 +1,239 @@
+import fs from 'fs-extra'
+import path from 'path'
+import os from 'os'
+import * as esbuild from 'esbuild'
+import * as url from 'url'
+import { Config } from '@tinacms/schema-tools'
+import * as dotenv from 'dotenv'
+
+const TINA_FOLDER = 'tina'
+const LEGACY_TINA_FOLDER = '.tina'
+const GENERATED_FOLDER = '__generated__'
+const GRAPHQL_JSON_FILE = '_graphql.json'
+const GRAPHQL_GQL_FILE = 'schema.gql'
+const SCHEMA_JSON_FILE = '_schema.json'
+const LOOKUP_JSON_FILE = '_lookup.json'
+
+export class ConfigManager {
+  config: Config
+  rootPath: string
+  tinaFolderPath: string
+  tinaConfigFilePath: string
+  tinaSpaPackagePath: string
+  envFilePath: string
+  generatedFolderPath: string
+  generatedGraphQLGQLPath: string
+  generatedGraphQLJSONPath: string
+  generatedSchemaJSONPath: string
+  generatedLookupJSONPath: string
+  generatedTypesTSFilePath: string
+  generatedTypesJSFilePath: string
+  generatedTypesDFilePath: string
+  generatedClientFilePath: string
+  generatedQueriesFilePath: string
+  generatedFragmentsFilePath: string
+  generatedQueriesAndFragmentsGlob: string
+  userQueriesAndFragmentsGlob: string
+  publicFolderPath: string
+  outputFolderPath: string
+  outputHTMLFilePath: string
+  outputGitignorePath: string
+  selfHostedDatabaseFilePath?: string
+  spaRootPath: string
+  spaHTMLPath: string
+
+  constructor(rootPath: string = process.cwd()) {
+    this.rootPath = rootPath.endsWith('/') ? rootPath.slice(0, -1) : rootPath
+  }
+
+  hasSelfHostedConfig() {
+    return !!this.selfHostedDatabaseFilePath
+  }
+
+  async processConfig() {
+    this.tinaFolderPath = await this.getTinaFolderPath(this.rootPath)
+
+    // TODO - .env should potentially be configurable
+    this.envFilePath = path.resolve(
```

Comment:
Is this needed? Could we just used the default behaviour of dotenv?

Reply:
I'm happy for it be a bit more explicit, if we need to open it up we can for some reason. This has the advantage of working when you're running the CLI from the monorepo and using `--rootPath`. So if it causes issues we can just use this conditionally.